### DISCUSSION TITLE
Add option to dump currently loaded keymap config

### DISCUSSION THREAD

**Discussion by noyez** [helix-editor/helix#10386] [Ideas]
As a method to add a self-documenting feature, believe it would helpful to add an option to dump the current keymap configuration (or maybe the whole configuration). This would serve 2 purposes
 1. give a user a starting point for a valid and properly formatted configuration with all options. 
 2. list all keymap configuration modes and options.

I use a similar option w/ zellij, namely `zellij setup --dump-config` will dump the whole configuration to stdout. This allows me to see all possible keymappings in a properly configured format.

**Comment by noyez** [comment]
> Excuse me, the official doc lacks what config?

For example, i wanted to remap `C-o` in windowed mode as i kept on hitting it by mistake. I looked over the docs to figure out how to do that, but i couldn't find it in the docs. If i am missing this, then please let me know. The goal if this issue is to offer a complete dump of the config so the user explore every config option. 

As an example my journey to remap `C-o` in window mode looked like the following, 

 i first tried
```
[keys.window]
C-o = "no_op"
```
But that lead to `Invalid mode 'window'`, 

then i tired 
```
[keys.windowed]
C-o = "no_op"
```
But that lead to `Invalid mode 'windowed'`, 

then i tried something like
```
[keys.normal]
C-o = "no_op"
```
That at least parsed, but didn't achieve my goal. 


After searching around i finally figured out the proper way to remap a key in windowed mode. 
```
[keys.normal.C-w]
o = "rotate_view"
C-o = "no_op"
```
I do not believe this is mentioned in the docs, and i don't think it needs to be mentioned since it is a niche keymap. However I believe if there was a way to dump the whole config to stdout or a file, a user would be able to see valid syntax for various modes and other options. I see it as a discoverability feature.

**Comment by kirawi** [comment]
> 1. give a user a starting point for a valid and properly formatted configuration with all options.

This was mentioned in https://github.com/helix-editor/helix/discussions/655. It sounds like an easy addition, but I'm not sure how the maintainers feel about it.
